graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +460,118 @@ def start_global_operations(self):
 
     def filter(self, predicate):
         """Execute a Filter Block."""
+        left_predicate_folded = isinstance(predicate.left, FoldedContextField)
+        right_predicate_folded = isinstance(predicate.right, FoldedContextField)
+        if self._in_fold or left_predicate_folded or right_predicate_folded:
+            raise NotImplementedError('Filters inside a fold are not implemented yet.')
+
         sql_expression = predicate.to_sql(self._aliases, self._current_alias)
         if self._is_in_optional_scope():
             sql_expression = sqlalchemy.or_(sql_expression,
                                             self._came_from[self._current_alias].is_(None))
         self._filters.append(sql_expression)
 
+    def _get_fold_join_descriptor_from_current_classname(self, fold_scope_location):
+        """Convert a fold location and preceding classname to a join descriptor from schema."""
+        # collect edge information to join the fold subquery to the main selectable
+        edge_direction, edge_name = fold_scope_location.fold_path[0]
+        full_edge_name = '{}_{}'.format(edge_direction, edge_name)
+        # only works if fold scope location is the immediate child of self._current_classname
+        return self._sql_schema_info.join_descriptors[self._current_classname][full_edge_name]
+
+    def _get_fold_join_info(self, fold_scope_location):
```

Comment:
What's the reason to not encapsulate this in the SQLFoldObject?

Reply:
It uses a lot of the folds from the CompilationState, including the schema, the query metadata table, and the current class name. On the other hand, it doesn't use any of the fields of the SQLFoldObject.  Seems like it might be more appropriate for that to be generated from CompilationState

Reply:
I still think it might be better to fully encapsulate all of the fold logic in SQLFoldObject. What do you think @bojanserafimov ?